sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala

```diff
@@ -233,7 +233,7 @@ case class GetArrayStructFields(
 case class GetArrayItem(
     child: Expression,
     ordinal: Expression,
-    failOnError: Boolean = SQLConf.get.ansiFailOnElementNotExists)
+    failOnError: Boolean = SQLConf.get.getConf(SQLConf.ANSI_STRICT_INDEX_OPERATOR))
```

Comment:
This `failOnError` looks like having a default value on its definition time. In that case, this PR is invalid.

Reply:
@cloud-fan 
I found the following in the log of failed test run:
```
[info]   Cause: org.apache.spark.SparkNoSuchElementException: Key brother does not exist. If necessary set spark.sql.ansi.strictIndexOperator to false to bypass this error.
```
I came to the same conclusion as @dongjoon-hyun pointed out above.
I think we can restore to renaming ansiFailOnElementNotExists in my original PR.